### ISSUE TITLE
Add gamepad support

### ISSUE THREAD

**Issue by wirelessdreamer** [matzman666/OpenVR-InputEmulator#13]
reading input from a usb gamepad (like the top shot elite controller) to emulate stuff like the touchpad would add a lot of flexibility to the system

here is a link to one of the controllers:

https://www.amazon.com/Cabelas-Top-Shot-Elite-Firearm-Controller/dp/B004FDYTI2/ref=pd_lpo_sbs_63_t_2?_encoding=UTF8&refRID=0CE7BCRARNWZ3SBHW9MB&th=1

**Comment by EVOL-LOVE** [comment]
Yes please add this feature. It would be awesome to send commands from VR treadmills back into the motion controllers. Please also support XInput and DirectInput as for example the Virtuix Omni is a DirectInput device. I do not know about Katwalk but I assume it is DirectInput as well.

**Comment by mmorselli** [comment]
@wirelessdreamer If you're still interested, I just got this working, a top shot elite with Vive Tracker, but can also work with a controller on top 

you need:

- JoyToKey (https://joytokey.net/en/)
to translate topshot buttons into keyboard keys

- AutoHotkey (https://autohotkey.com/)
to execute a script on a key press

- OpenVR-InputEmulator Command-Line Client
to send input to OpenVR controller

script:

trigger.bat (put in the Command-Line Client folder)

```
client_commandline.exe axisevent 3 1 1 0
client_commandline.exe axisevent 3 1 0 0
```
change "3" with your controller ID

topshot2openvr.ahk (AutoHotKey script, put in the Command-Line Client folder)

```
#NoEnv  
SendMode Input  
SetWorkingDir %A_ScriptDir%  
f::
Process, Exist, client_commandline.exe
If Not ErrorLevel
    Run trigger.bat,,hide
return

```
assuming "f" as the key associated with the trigger in JoyToKey

works flawlessly with zero lag

here you can find a 3D Print file of a Vive controller mount  https://www.thingiverse.com/thing:2633912


Of course, integrated support of XInput in OpenVR-InputEmulator would be a lot better



![topshotelite](https://user-images.githubusercontent.com/5322634/37870551-284d2496-2fd9-11e8-88fc-bae141d19bc3.jpg)
